### PR TITLE
Task 4: Connect Lamdas to DB

### PR DESCRIPTION
**Pull request by IvanPresmytsky**
**What was done:**

- DB initialization is implemented at the beginning of lamdas (if DB is empty)
- Integrate GET/products lambda to return a list of products from the database (variables for DB was not committed and stored into .env file)
- Implement POST/products lambda and implement its logic so it will be creating a new item in a products table.
- The Frontend application is integrated with product service (/products API) and products from product-service are available on FE
- POST/products lambda functions return error 400 status code if product data is invalid
- All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- All lambdas do console.log for each incoming requests and their arguments
- Transaction based creation of the product is implemented (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end-user and vice versa)

**Links to lamdas:**
GET - https://2s5xprqt4k.execute-api.eu-west-1.amazonaws.com/dev/products
POST - https://2s5xprqt4k.execute-api.eu-west-1.amazonaws.com/dev/products
GET - https://2s5xprqt4k.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}

**Link to a working Front-End application:** https://d13u81eg1opkn3.cloudfront.net/

### CODE REVIEW

**Comment by shurygindv** [comment]
"World of BeerCraft!" 9/9 points )

but seems, POST /products doesn't work very well, when sending such model:

{
**"count"**: 344,
**"description"**: "some content",
**"price"**: 777,
**"title"**: "some title"
}

and got `Badrequest!"id",
"title",
"description",
priceandcountparametersshouldbeprovided!`

File: product-service/utils/queryBuilders/insertProductsQueryBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import products from '../../mocks/products.json';
+import { TProduct } from '../../types';
+
+export const getInsertProductsQuery = (productsToInsert = products) => {
```
Comment (SkorykV) [inline product-service/utils/queryBuilders/insertProductsQueryBuilder.ts:4]: this query builders are not used, are they?

**Comment by SkorykV** [comment]
I liked messages builder conception!
but seems that you didn't integrate products creation in frontend? Also I didn't understand why id is a required param  for post :)

File: product-service/utils/queryBuilders/insertProductsQueryBuilder.ts (hunk 1 above)
Comment (IvanPresmytsky) [reply product-service/utils/queryBuilders/insertProductsQueryBuilder.ts:4]: These query builders are used inside DB client.

**Comment by IvanPresmytsky** [comment]
> I liked messages builder conception!
> but seems that you didn't integrate products creation in frontend? Also I didn't understand why id is a required param for post :)

Thanks for your review! I haven't integrated product creation to FE yet. It was not a requirement for this task, but I'm going to do it in the next PR. I know the issue with the required 'id' I will fix it in the next PR also.

**Comment by IvanPresmytsky** [comment]
> "World of BeerCraft!" 9/9 points )
> 
> but seems, POST /products doesn't work very well, when sending such model:
> 
> {
> **"count"**: 344,
> **"description"**: "some content",
> **"price"**: 777,
> **"title"**: "some title"
> }
> 
> and got `Badrequest!"id", "title", "description", priceandcountparametersshouldbeprovided!`

Thanks for your review! I will fix the issue with 'id' in the next PR.

**Pull request merged**